tablecodec/tablecodec.go

```diff
@@ -227,16 +227,15 @@ func DecodeColumnValue(data []byte, ft *types.FieldType, loc *time.Location) (ty
 	return colDatum, nil
 }
 
-// DecodeRow decodes a byte slice into datums.
+// DecodeRowWithMap decodes a byte slice into datums with a existing row map.
 // Row layout: colID1, value1, colID2, value2, .....
-func DecodeRow(b []byte, cols map[int64]*types.FieldType, loc *time.Location) (map[int64]types.Datum, error) {
-	if b == nil {
+func DecodeRowWithMap(b []byte, cols map[int64]*types.FieldType, loc *time.Location, row map[int64]types.Datum) (map[int64]types.Datum, error) {
+	if row == nil || b == nil {
```

Comment:
How about creating the map if `row` is nil at line 239?
Then we don't need to create the map in `DecodeRow`.

Or we just remove `DecodeRow`, use `DecodeRowWithMap` instead?
Or adding an argument `row` to `DecodeRow`?